sql/src/main/java/io/crate/protocols/postgres/types/PGArray.java

```diff
@@ -95,11 +95,17 @@ public Object readBinaryValue(ChannelBuffer buffer, int valueLength) {
                     }
                 } else {
                     bytes = innerType.encodeAsUTF8Text(o);
-                    encodedValues.add((byte) '\"');
+                    boolean isJson = JsonType.OID == innerType.oid();
+
+                    encodedValues.add((byte) '"');
                     for (byte aByte : bytes) {
+                        // Escape double quotes with backslash for json
+                        if (isJson && (char) aByte == '"') {
```

Comment:
You mean something like that?
`if (isJson) {
    for (byte aByte : bytes) {
        // Escape double quotes with backslash for json
        if ((char) aByte == '"') {
            encodedValues.add((byte) '\\');
        }
        encodedValues.add(aByte);
    }
} else {
    for (byte aByte : bytes) {
        encodedValues.add(aByte);
    }
}`